src/program/mixins.py

```diff
@@ -39,42 +36,33 @@ def dispatch(self, request, *args, **kwargs):
                 reverse("program:proposal_list", kwargs={"camp_slug": self.camp.slug}),
             )
 
-        # alright, continue with the request
-        return super().dispatch(request, *args, **kwargs)
-
 
 class EnsureWritableCampMixin:
-    def dispatch(self, request, *args, **kwargs):
+    def setup(self, request, *args, **kwargs):
         # do not permit view if camp is in readonly mode
         if self.camp.read_only:
             messages.error(request, "No thanks")
             return redirect(
```

Comment:
we cannot return a HttpResponse from inside setup(), only dispatch() can do that. Either raise an exception or keep this as dispatch()